akka-remote/src/test/scala/akka/remote/DaemonicSpec.scala

```diff
@@ -38,17 +27,23 @@ class DaemonicSpec extends AkkaSpec {
         akka.log-dead-letters-during-shutdown = off
       """))
 
-      val unusedAddress = getOtherAddress(daemonicSystem, "tcp").copy(port = Some(unusedPort))
-      val selection = daemonicSystem.actorSelection(s"${unusedAddress}/user/SomeActor")
-      selection ! "whatever"
-      Thread.sleep(2.seconds.dilated.toMillis)
+      try {
+        val unusedPort = 86 // very unlikely to ever be used, "system port" range reserved for Micro Focus Cobol
+
+        val unusedAddress = RARP(daemonicSystem).provider.getExternalAddressFor(Address(s"akka.tcp", "", "", unusedPort)).get
+        val selection = daemonicSystem.actorSelection(s"$unusedAddress/user/SomeActor")
+        selection ! "whatever"
 
-      // get new non daemonic threads running
-      val newNonDaemons: Set[Thread] = Thread.getAllStackTraces().keySet().asScala.seq.
-        filter(t ⇒ !origThreads(t) && t.isDaemon == false).to[Set]
+        // get new non daemonic threads running
+        awaitAssert({
+          val newNonDaemons: Set[Thread] = Thread.getAllStackTraces.keySet().asScala.seq.
+            filter(t ⇒ !origThreads(t) && !t.isDaemon).to[Set]
+          newNonDaemons should ===(Set.empty[Thread])
+        }, 4.seconds.dilated)
```

Comment:
that is already dilated by `awaitAssert`

Reply:
D'oh. I never remember where it is and where it isn't...